graph/graph.go

```diff
@@ -234,6 +241,28 @@ func checkVisibility(target labels.Label, visibilities []string) bool {
 	return false
 }
 
+// isSubpackage returns true if pkg is the name of a package whose BUILD file would exist within the
+// directory tree belonging to the package basePkg, or false otherwise.
+//
+// A basePkg consisting of the empty string denotes the top-level package; given the definition of
+// "sub-package" above, it follows that all packages are sub-packages of the top-level package.
+func isSubpackage(basePkg, pkg string) bool {
+	if basePkg == "" {
+		return true
+	}
+	basePkgParts := strings.Split(basePkg, "/")
+	pkgParts := strings.Split(pkg, "/")
+	if len(basePkgParts) > len(pkgParts) {
+		return false
+	}
+	for i := range basePkgParts {
+		if basePkgParts[i] != pkgParts[i] {
```

Comment:
`return slices.Equal(pkgParts, basePkgParts[:len(pkgParts)])`

alternatively, skip all the splitting and `return basePkg == pkg || strings.HasPrefix(pkg, basePkg + "/")`

Reply:
`basePkg == pkg || strings.HasPrefix(pkg, basePkg + "/")` won't work in the case where `basePkg` is the empty string (i.e. the top-level package) and `pkg` is something other than the top-level package. The slices are the wrong way around in the latter suggestion, but that's a much nicer alternative to the `for` loop, thanks!